plugins/janus_sip.c

```diff
@@ -1385,6 +1400,29 @@ static void janus_sip_remove_quotes(char *str) {
 	}
 }
 
+static json_t *janus_sip_get_incoming_headers(const sip_t *sip, const janus_sip_session *session) {
+	json_t *headers = json_object();
+	if(!sip)
+		return headers;
+	sip_unknown_t *unknown_header = sip->sip_unknown;
```

Comment:
Any reason why you're only traversing the unknown headers? I was under the assumption that this could be used to intercept any header, but then it wasn't returning anything when I attempted to intercept some common headers (e.g., "User-Agent") and noticed this. I guess this is what you mean by "custom" headers in the documentation, but it's actually not clear at all. As such, it should either be made more explicit in the documentation, or the `janus_sip_get_incoming_headers` implementation should be extended to any header, whether it's standard or not (which would sound more natural to me).

Reply:
After looking at the Sofia SIP [structure for messages](http://sofia-sip.sourceforge.net/refdocs/sip/structsip__s.html) in the docs, though, extending this to all headers seems pretty much impossible... Known headers seem to be all parsed to their own structure, and not available as a key/value list anywhere else like unknown headers are in `sip_unknown`. If so, then I'd just recommend to make it more explicit in the docs (and PR title) that the new feature only impacts non-standard headers.

Reply:
Yes, you are right. I will update documentation.